subsys/pm/device_runtime.c

```diff
@@ -67,7 +67,7 @@ static int runtime_suspend(const struct device *dev, bool async)
 		goto unlock;
 	}
 
-	if (async && !k_is_pre_kernel()) {
+	if ((async && !k_is_pre_kernel()) || k_is_in_isr()) {
```

Comment:
we can obviate k_is_in_isr() here, since async == true when in using _async from an ISR

Reply:
yeah, i left that anyway just in case someone compiles the kernel with asserts off.

Reply:
but that modifies the sync path, right?